api/definitions/comment_rejection_object.md

```diff
@@ -0,0 +1,50 @@
+# Comment Rejection Object
+
+*Note: Info here isn't 100% complete, since we don't have sample information for what most comment rejection objects actually look like.*
+
+A comment rejection object looks like this:
+
+```
+{
+    rejected: /* Rejection code - see below */
+    status: /* (optional) Status object with rejection details - see below */
+    appealId: /* (if IP muted) Case number to contact ST with to appeal block */
+}
+```
+
+The `rejected` property will be one of the following:
+
+- `"isEmpty"`: the comment is empty
+- `"isFlood"`: sending too many comments in a short period of time
+- `"isBad"`: content was caught by the filter bot
+- `"hasChatSite"`: content contained a link to a website with unmoderated chat
+- `"isSpam"`: same content has been posted too many times
+- `"isMuted"`: posted too many bad comments and account has been muted for the day
+- `"isUnconstructive"`: content was detected as being mean or disrespectful
+- `"isDisallowed"`: tried to post on a page where comments are disabled
+- `"isIPMuted"`: comment sent from a network blocked for posting the community guidelines too many times
+- `"isTooLong"`: content surpasses maximum comment length
+- `"isNotPermitted"`: user email address hasn't been confirmed yet
+
+The `status` property, if present, may have a `mute_status` property. This object will look like this:
+
+```
+{
+    offenses: /* Array of offense objects */
+    showWarning: /* Override to always show warning regardless of offense list */
+    muteExpiresAt: /* When the user will be unmuted (*1000 to convert to Unix timestamp) */
+    currentMessageType: /* Latest(?) offense message type */
+}
+```
+
+Offense objects look like this:
+
+```
+{
+    createdAt: /* When the offense was caught (*1000 to convert to Unix timestamp) */
+    expiresAt: /* When the offense expires (*1000 to convert to Unix timestamp) */
+    messageType: /* What kind of offense was caught */
+}
+```
+
+For example, an offense object might indicate it was caught today, expires in one week, and has the message type `"sexual"`. The `mute_status` it is part of may indicate the mute itself expires in only five minutes.
```

Comment:
Do we have any other examples of offense object message types?

Reply:
Nope, this is all I gathered.

> (Also one experimental case, though I certainly don't advocate for anyone else to intentionally run into error messages to gather more data themselves. :P)

I tried posting a bad word once to see what the object looked like, but should probably have just constructed one up from the reference code in scratch-www: https://github.com/LLK/scratch-www/blob/183a31d5ed8e7fcb1633caabd3a0928a6936a653/src/views/preview/comment/compose-comment.jsx#L95-L122